sql/src/test/java/io/crate/operation/projectors/ProjectingBatchConsumerTest.java

```diff
@@ -0,0 +1,167 @@
+import io.crate.operation.InputFactory;
+import io.crate.operation.operator.EqOperator;
+import io.crate.planner.projection.FilterProjection;
+import io.crate.planner.projection.GroupProjection;
+import io.crate.types.DataTypes;
+import org.elasticsearch.action.bulk.BulkRetryCoordinatorPool;
+import org.elasticsearch.cluster.ClusterService;
+import org.elasticsearch.cluster.metadata.IndexNameExpressionResolver;
+import org.elasticsearch.common.breaker.CircuitBreaker;
+import org.elasticsearch.common.breaker.NoopCircuitBreaker;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Answers;
+import org.mockito.MockitoAnnotations;
+
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+import static io.crate.testing.TestingHelpers.getFunctions;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Mockito.mock;
+
+public class ProjectingBatchConsumerTest {
+
+    private static final RamAccountingContext RAM_ACCOUNTING_CONTEXT =
+        new RamAccountingContext("dummy", new NoopCircuitBreaker(CircuitBreaker.FIELDDATA));
+    private Functions functions;
+    private ThreadPool threadPool;
+    private ProjectorFactory projectorFactory;
+
+
+    @Before
+    public void prepare() {
+        MockitoAnnotations.initMocks(this);
+        functions = getFunctions();
+        threadPool = new ThreadPool("testing");
+        projectorFactory = new ProjectionToProjectorVisitor(
+            mock(ClusterService.class),
+            functions,
+            new IndexNameExpressionResolver(Settings.EMPTY),
+            threadPool,
+            Settings.EMPTY,
+            mock(TransportActionProvider.class, Answers.RETURNS_DEEP_STUBS.get()),
+            mock(BulkRetryCoordinatorPool.class),
+            new InputFactory(functions),
+            EvaluatingNormalizer.functionOnlyNormalizer(functions, ReplaceMode.COPY),
+            null
+        );
+    }
+
+    @After
+    public void after() throws Exception {
+        threadPool.shutdown();
+        threadPool.awaitTermination(1, TimeUnit.SECONDS);
+    }
+
+    @Test
+    public void consumerRequiresScrollAndProjectorsDontSupportScrolling() throws Exception {
+        EqOperator op = (EqOperator) functions.get(
+            new FunctionIdent(EqOperator.NAME, ImmutableList.of(DataTypes.INTEGER, DataTypes.INTEGER)));
+        Function function = new Function(op.info(), Arrays.asList(Literal.of(2), new InputColumn(1)));
+        FilterProjection filterProjection = new FilterProjection(function,
+            Arrays.asList(new InputColumn(0), new InputColumn(1)));
+
+        BatchConsumer delegateConsumerRequiresScroll = new BatchConsumer() {
```

Comment:
Would probably be worth to create a dummy class in the test that can be used in all tests here